website/docs/docs/deploy/deploy-environments.md

```diff
@@ -186,6 +190,37 @@ This section allows you to determine the credentials that should be used when co
 </WHCode>
 
 
+## Staging environment <Lifecycle status='beta' />
+
+:::note
+Currently in limited availability beta.
+:::
+
+Staging environments are a useful way to grant developers access to deployment workflows and tools while limiting access to production data. Staging environments are configured with their own, long living branch (for example, `staging`) that may be very similar to `main` in many ways, but limits the data that the developers can access. 
```

Comment:
> that may be very similar to `main` in many ways

Usually _ahead_ of `main`, with staged releases from `staging` to `main`. There's a good overview of how this all works here: https://docs.getdbt.com/blog/the-case-against-git-cherry-picking

> but limits the data that the developers can access.

The mechanism for this is different. This needs to be applied **within** the data warehouse, by configuring Staging to point to:
- a different data warehouse instance entirely (different Snowflake account, BigQuery project, etc) — via env vars or Extended Attributes
- a different set of source schemas (e.g. using env vars within project code definitions of `sources`)